### ISSUE TITLE
Black Screen after sleep with your EFI files

### ISSUE THREAD

**Issue by Jialin-Wu** [romancin/Dell-XPS-7590-OpenCore#5]
Hi, Romancin,

  Actually, I have tried a lot of EFI batch files from GitHub repositories but only yours could drive my system and completed MACOS installation on my XPS 7590 OLED 4K non-touch screen + 1820A WiFi adaptor. Appreciate with your sharing. 
  One of remained issue is black screen after sleep on MacOS 10.15.6 or 10.15.7. I think that the laptop didn't really go to sleep since I could hear the sounds of warning when I click laptop's keyboard.  After sleeping happened on my laptop, the screen is black unless to press the power button and reboot the system by manual. I have tried to up to date these kext drivers and DDST files but no improvement. 
  Another issue is laptop's screen brightness is not working. Even I press " Fn+ brightness button". I could see the icon is pop-up but screen brightness doesn't change anymore.
   Would you pls. take a look it? Thanks in advance.

**Comment by CryptoNeverSleeps** [comment]
OLED's are a pain... I love the screen..... This is the only thing keeping us with the OLED model from 100% working.

**Comment by Jialin-Wu** [comment]
> OLED's are a pain... I love the screen..... This is the only thing keeping us with the OLED model from 100% working.

Exactly, I suppose that both brightness and sleeping issues are related with this "lovely" 4K OLED.  Not sure if next MacOS Version have improved to not.  Maybe 1080P screen have no issues on them.

**Comment by CryptoNeverSleeps** [comment]
1080P and 4K Touch Models of the 7590 Work with no problem at all... Our issue is 100% related to the OLED Screen.

**Comment by romancin** [comment]
So sorry, but I have a 4K IPS model and cannot try with OLED screen. I will leave this issue open in order to someone can help here.

**Comment by ZeroShadowX** [comment]
Hi romancin,

I have the 4K IPS model just like you and it seems after the 10.15.7 update that the black screen after sleep is also present on my machine. Didn't have any issues on 10.15.6 everything worked perfectly. 

I havent had a chance to do anything testing but can confirm this happens for me as well. Probably a kext or a SSDT needs to be updated but not sure where to start to fix it. 

Will be more than happy to test anything you might need.

**Comment by CryptoNeverSleeps** [comment]
I'm actually about to purchase the 4K Touch on eBay and replace my screen....  Not sure if enough people in the community see this as an issue......

**Comment by CryptoNeverSleeps** [comment]
Look into 4K IPS fix in latest lilu whatevergreen... there was a issue with 4K 48Hz / 60Hz workaround thats fixed in the latest lilu whatevergreen kext.

**Comment by Jialin-Wu** [comment]
have you tried the latest whatevergreen kext in your MacOS or not? Any improvement on it?

**Comment by Jialin-Wu** [comment]
Hi, 

I found out that Shady.App could run on the latest Mac OS - Big Sur 11.4 and "change" the brightness of OLED. Shady will add shadow addon layer in MAC desktop and It's workaround. The sleeping issue isn't resolved and I suppose that we will get improvement after apple production start to use OLED screen in the future.
